src/components/modals/CreateGroupsOrTypesModal.svelte

```diff
@@ -0,0 +1,210 @@
+    if (newTypeName === '') {
+      newTypeError = 'Please enter a new type name.';
+    } else {
+      effects.createExternalEventType({ name: newTypeName }, user);
+      newTypeName = '';
+    }
+  }
+</script>
+
+<Modal height={240} width={600}>
+  <ModalHeader on:close>Create Derivation Groups or Types</ModalHeader>
+  <ModalContent>
+    <CssGrid columns={modalColumnSize} minHeight="100%">
+      <div class="derivation-groups-modal-filter-container">
+        <div style="width:100%">
+          <Tabs class="creation-tabs" tabListClassName="creation-tabs-list">
+            <svelte:fragment slot="tab-list">
+              <Tab class="creation-tab">Derivation Group</Tab>
+              <Tab class="creation-tab">External Source Type</Tab>
+              <Tab class="creation-tab">External Event Type</Tab>
+            </svelte:fragment>
+            <TabPanel>
+              <p style="padding-top:10px">Provide a name and an external source type for the new derivation group.</p>
+              <p><i>The newly created group will be empty, though you can upload sources into it.</i></p>
+              <div class="timeline-editor-layer-filter">
+                <input
+                  bind:value={newTypeName}
+                  on:change={() => (newTypeError = null)}
+                  autocomplete="off"
+                  class="st-input w-100"
+                  name="filter-ee"
+                  placeholder={'New Derivation Group Name'}
+                />
+                <select
+                  bind:value={newTypeSourceType}
+                  on:change={() => (newTypeError = null)}
+                  name="newTypeSourceType"
+                  class="st-select"
+                  style="width: 200px"
+                >
+                  {#each $externalSourceTypes as sourceType}
+                    <option value={sourceType.name}>{sourceType.name}</option>
+                  {/each}
+                </select>
+                <button class="st-button w-10" type="submit" on:click|preventDefault={onCreateDerivationGroup}>
+                  Create
+                </button>
+              </div></TabPanel
+            >
+            <TabPanel>
+              <p style="padding-top:10px">Provide a name for the new external source type.</p>
+              <p><i>The newly created external source type will be empty, though you can upload sources into it.</i></p>
+              <div class="timeline-editor-layer-filter">
+                <input
+                  bind:value={newTypeName}
+                  on:change={() => (newTypeError = null)}
+                  autocomplete="off"
+                  class="st-input w-100"
+                  name="filter-ee"
+                  placeholder={'New External Source Type Name'}
+                />
+                <button class="st-button w-10" type="submit" on:click|preventDefault={onCreateExternalSourceType}>
+                  Create
+                </button>
+              </div></TabPanel
```

Comment:
Can `</TabPanel>` be on a new line? I think the autoformatter did something weird here.